### DISCUSSION TITLE
Programatically eager load a lazy loaded component?

### DISCUSSION THREAD

**Discussion by that-one-arab** [vercel/next.js#69782] [Help]
## Context

I have a list of steps in a stepper.

The stepper has 2 steps.

The 1st step is normally imported and loaded directly

The 2nd step is lazy loaded using `next/dynamic`

## Problem

Once the user moves to the 2nd step, only then would the 2nd step be loaded, causing a small delay, which is to be expected.

## Suggestion

Ideally, I would like for the 2nd step to load directly after the 1st step is loaded.

That way, the 2nd step is loaded in the background as the user is filling the form of the 1st step, once he moves on to the 2nd step it would be already loaded without any delay.

How can I implement this behaviour?

**Comment by icyJoseph** [comment]
In App Router, I think this can be tricky... Somehow `ReactDOM.preload` has to be used but I am not sure how.

In Pages Router, the Webpack Magic Comments, were the way to go:

- https://webpack.js.org/api/module-methods/#webpackprefetch
- https://webpack.js.org/api/module-methods/#webpackpreload

```js
import(
  /* webpackPrefetch: true */
  /* webpackPreload: true */
  `./locale/${language}`
);

```

However, I noticed that, webpackChunkName doesn't work in App Router, so I reckon the others don't work either.
